### DISCUSSION TITLE
How can I catch all the unhandled errors that might occur in getServerSideProps() functions?

### DISCUSSION THREAD

**Discussion by lumenwrites** [vercel/next.js#40536] [Help]
Hi! I want my app to have good error handling. Whenever something throws an error inside getServerSideProps() function (for example, I tried to fetch some data from the database and failed), I want to log this error to a logging platform that I'm using (Axiom), so that I could be notified about it and debug it. And I want to show an error page to the user.

But I don't want to manually wrap every little thing that might thrown an error in a try-catch block, inside every getServerSideProps() function.

So I'm thinking, would it be okay to create a wrapper that would wrap all the getServerSideProps() functions, and catch all the unhandled errors? Something like this:

```javascript
async function getSSPWrapper(context, getSSP) {
  try {
    return await getSSP(context)
  } catch (e) {
    const error = { message: e.message, stack: e.stack }
    logServer.error(`Unhandled error: ${e.message}`, { error })
    return { props: { errorCode: 500 } }
  }
}
```

And then inside my pages do something like this:
```javascript
const getSSP: GetServerSideProps = async ({ req, res }) => {
  // The code that would normally go into getServerSideProps

  // Some function that might fail and throw some error:
  throw new Error('Something went wrong in getSSP')

  return { props: {} }
}

export const getServerSideProps = (context) => getSSPWrapper(context, getSSP)
```

That way all the unhandled errors inside of getServerSideProps() functions would get caught and logged out by getSSPWrapper().

This seems to make sense in my head, but I've never seen anyone doing things like this. Is this a good idea? Is there a better way?

**Comment by gaston-flores** [comment]
I think it might be easier to add your logging logic to [getInitialProps in the `_error` page](https://nextjs.org/docs/advanced-features/custom-error-page#more-advanced-error-page-customizing). You'll get access to the original error, and you can show a custom error page.
